gramex/apps/admin2/gramexadmin.py

```diff
@@ -60,7 +83,17 @@ class AdminFormHandler(gramex.handlers.FormHandler):
     def setup(cls, **kwargs):
         # admin_kwargs.authhandler is a url: key that holds an AuthHandler. Get its kwargs
         try:
-            authhandler, auth_conf, data_conf = get_auth_conf(kwargs.get('admin_kwargs', {}))
+            admin_kwargs = kwargs.get('admin_kwargs', {})
+            if kwargs.get('rules', False):
+                # Get the rules for formhandler
+                authhandler = admin_kwargs.get('authhandler', False)
+                if not authhandler:
+                    raise ValueError('Missing authhandler.')
```

Comment:
Mention the url: key in the error message to identify this better. It might be cls.name. See get_auth_conf.